.github/workflows/codeql.yml

```diff
@@ -0,0 +1,92 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master", "develop" ]
+  pull_request:
+    branches: [ "master", "develop" ]
+  schedule:
+    - cron: '27 10 * * 2'
```

Comment:
i don't think it makes sense to run it as a scheduled cron, pushing to branches should be sufficient or can you also schedule it as a test for when a PR is opened, before the merge, like a check it needs to pass before you can even do a merge.

Reply:
The CodeQL is also getting updated weekly. So new vulnerabilities might get introduced on the same code. It might be useful to let it check every now and then.